perfmetrics/scripts/hns_rename_folders_metrics/renaming_benchmark.py

```diff
@@ -97,6 +97,50 @@ def _record_time_for_folder_rename(mount_point,folder,num_samples):
   return time_op
 
 
+def _record_time_of_operation(mount_point, dir, num_samples):
+  """
+  This function records the time of rename operation for each testing folder
+  inside dir directory,for num_samples number of test runs.
+
+  Args:
+    mount_point: Mount point for the GCS bucket.
+    dir: JSON object representing the folders structure.
+    num_samples: Number of samples to collect for each test.
+
+  Returns:
+    A dictionary with lists containing time of rename operations in seconds,
+    corresponding to each folder.
+  """
+  results = dict()
+  # Collecting metrics for non-nested folders.
+  for folder in dir["folders"]["folder_structure"]:
+    results[folder["name"]] = _record_time_for_folder_rename(mount_point,folder,num_samples)
+  #TODO Add metric collection logic for nested-folders
+  return results
+
+
+def _perform_testing(dir, test_type, num_samples, results):
+  """
+  This function performs rename operations and records time of operation .
+  """
+  if test_type == "flat":
+    # Mounting the flat bucket.
+    flat_mount_flags = "--implicit-dirs --rename-dir-limit=1000000"
+    flat_bucket = mount_gcs_bucket(dir["name"], flat_mount_flags, log)
+
+    # Record time of operation and populate the results dict.
+    flat_results = _record_time_of_operation(flat_bucket, dir, num_samples)
+    results["flat"] = flat_results
+
+    unmount_gcs_bucket(dir["name"], log)
+  elif test_type == "hns":
```

Comment:
Default can be flat, In the start of method we can add if check for hns, there is no need for else, return from inside if hns.

Reply:
Updated.